docs/src/pages/components/data-grid/demo/FullFeauredDemo.tsx

```diff
@@ -0,0 +1,153 @@
+
+  const onSizeChange = React.useCallback((e) => {
+    setSize(Number(e.target.value));
+  }, []);
+
+  const onDatasetChange = React.useCallback((e) => {
+    setType(e.target.value);
+  }, []);
+
+  const onPaginationChange = React.useCallback((e) => {
+    setSelectedPaginationValue(e.target.value);
+  }, []);
+
+  const applyChanges = React.useCallback(() => {
+    onApply({ size: sizeState, type: typeState, pagesize: selectedPaginationValue });
+  }, [sizeState, typeState, selectedPaginationValue, onApply]);
+
+  return (
+    <FormGroup className="MuiFormGroup-options" row>
+      <FormControl size="small" component="fieldset">
+        <FormLabel component="legend">Dataset</FormLabel>
+        <Select value={typeState} onChange={onDatasetChange}>
+          <MenuItem value="Employee">Employee</MenuItem>
+          <MenuItem value="Commodity">Commodity</MenuItem>
+        </Select>
+      </FormControl>
+      <FormControl size="small" component="fieldset">
+        <FormLabel component="legend">Rows</FormLabel>
+        <Select value={sizeState} onChange={onSizeChange}>
+          <MenuItem value={100}>100</MenuItem>
+          <MenuItem value={1000}>{Number(1000).toLocaleString()}</MenuItem>
+          <MenuItem value={10000}>{Number(10000).toLocaleString()}</MenuItem>
+          <MenuItem value={100000}>{Number(100000).toLocaleString()}</MenuItem>
+        </Select>
+      </FormControl>
+      <FormControl size="small" component="fieldset">
+        <FormLabel component="legend">Page Size</FormLabel>
+        <Select value={selectedPaginationValue} onChange={onPaginationChange}>
+          <MenuItem value={-1}>off</MenuItem>
+          <MenuItem value={0}>auto</MenuItem>
+          <MenuItem value={25}>25</MenuItem>
+          <MenuItem value={100}>100</MenuItem>
+          <MenuItem value={1000}>{Number(1000).toLocaleString()}</MenuItem>
+        </Select>
+      </FormControl>
+      <Button size="small" variant="outlined" color="primary" onClick={applyChanges}>
+        <KeyboardArrowRightIcon fontSize="small" /> Apply
+      </Button>
+    </FormGroup>
+  );
+};
+
+export default function FullFeaturedDemo() {
+  const classes = useStyles();
+  const [type, setType] = React.useState('Commodity');
+  const [size, setSize] = React.useState(100);
+  const { data } = useDemoData({
+    dataSet: type as any,
+    rowLength: size,
+    maxColumns: 20,
+  });
+  const [gridData, setGridData] = React.useState({});
+  const [pagination, setPagination] = React.useState({});
+
+  const onApplyClick = async (settings) => {
+    if (size !== settings.size) {
+      setSize(settings.size);
+    }
+
+    if (type !== settings.type) {
+      setType(settings.type.toLowerCase());
+    }
+
+    const newData = await getRealData(
+      settings.size,
+      settings.type === 'Commodity' ? getCommodityColumns() : getEmployeeColumns(),
+    );
+    setGridData(newData);
```

Comment:
This is unsafe. It could lead to setState called in an unmounted component

Reply:
This should handle the case, I didn't want to wrap it in a cantabile promise.